core/scripts/liquidity-mining/CalculateBalancerLPProviders.js

```diff
@@ -135,23 +166,33 @@ async function _updatePayoutAtBlock(blockNumber, shareHolderPayout, bPool) {
 }
 
 // Generate a json file containing the shareholder output address and associated $UMA token payouts.
-function _saveShareHolderPayout(shareHolderPayout) {
+function _saveShareHolderPayout(shareHolderPayout, week) {
   // First, clean the shareHolderPayout of all zero recipients and convert from wei scaled number.
   for (shareHolder of Object.keys(shareHolderPayout)) {
     if (shareHolderPayout[shareHolder] == "0") delete shareHolderPayout[shareHolder];
     else shareHolderPayout[shareHolder] = fromWei(shareHolderPayout[shareHolder]);
   }
 
-  const savePath = `${path.resolve(__dirname)}/weekly-payouts/${argv.week}_week_UMAsToDistribute.json`;
+  const savePath = `${path.resolve(__dirname)}/weekly-payouts/${week}_week_UMAsToDistribute.json`;
   fs.writeFile(savePath, JSON.stringify(shareHolderPayout), err => {
     if (err) return console.error(err);
     console.log("🗄  File successfully written to", savePath);
   });
 }
 
-main()
-  .then(() => {})
-  .catch(err => {
-    console.error(err);
-    process.exit(1);
-  });
+// Function with a callback structured like this is required to enable `truffle exec` to run this script.
```

Comment:
Why did you decide to change this to using truffle exec?

Reply:
Yup. exactly that. Ideally, there is a setup where you can:
1. run using `node ...`
2. run using `truffle exec ...`
3. test using `truffle test ...`
I was however only able to get exclusively 1 to work or 2 & 3 to work. I'm sure there is a way to export this process such that it can be achieved but I was not able to figure this out so converted back to our old pattern that is bound to truffle :/